apk-generator/v1/main.js

```diff
@@ -27,112 +7,134 @@ var config = {
-        ajaxCall1();
-        $('#submit').prop("disabled", true);
-
-        function ajaxCall1() {
-            updatePercentUpload(35);
-            $.ajax({
-                type: "POST",
-                url: "/runPy.php",
-                data: {
-                    timestamp: timestamp
-                },
-                success: function(response) {
-                    console.log("Success", response);
-                    updatePercentUpload(100);
-                    $('#download').prop("disabled", false);
-                    $('#download').click(function() {
-                        window.location = response;
-                    });
-                    document.getElementById("status").innerHTML = "App build completed!";
-                }
-            });
-        }
-    });
-}
-}); //after this
-function updatePercentUpload(perc) {
-    $('#upload-progress').css('width', perc + '%');
-    $('#upload-progress').html(parseInt(perc) + '%');
-}
-
-function uploadFile() {
-    var file_data = $('#uploadZip').prop('files')[0];
-    var form_data = new FormData();
-    form_data.append('file', file_data);
-    form_data.append('timestamp', timestamp);
-    ajax.upload.addEventListener("progress", progressHandler, false);
-    ajax.addEventListener("load", completeHandler, false);
-    ajax.open("POST", "upload.php");
-    ajax.send(form_data);
-
-}
-
-function progressHandler(event) {
-    var percent = (event.loaded / event.total) * 100;
-    updatePercentUpload(Math.round(percent))
-    document.getElementById("status").innerHTML = Math.round(percent) + "% uploaded... please wait";
-}
-
-function completeHandler(event) {
-    document.getElementById("status").innerHTML = "Upload Complete, click on the button below to build your app";
-    $('#submit').prop("disabled", false);;
-}
-
-function checkemail(emailse) {
-    var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
-    return re.test(emailse);
-}
\ No newline at end of file
+    else {
+        updatePercentUpload(20);
+        var fileData = $("#uploadZip").prop("files")[0];
+        var formData = new FormData();
+        formData.append("file", fileData);
+        firebase.database().ref("users/" + timestamp).set(obj);
+        database.ref("users/" + timestamp).once("value").then(function (snapshot) {
+            updatePercentUpload(10);
+            console.log("Received value", snapshot.val());
+            document.getElementById("status").innerHTML = "Building the app. This might take a while...";
+            function ajaxCall1() {
+                updatePercentUpload(35);
+                $.ajax({
+                    type: "POST",
+                    url: "/runPy.php",
+                    data: {
+                        timestamp
+                    },
+                    success(response) {
```

Comment:
Actually this issues were already in code. But it is showing here because some functions were used before they are defined and i have moved them before function using it.